go-controller/pkg/cni/cnishim.go

```diff
@@ -94,6 +98,42 @@ func (p *Plugin) doCNI(url string, req interface{}) ([]byte, error) {
 	return body, nil
 }
 
+func setupLogging(conf *ovntypes.NetConf) {
+	var err error
+	var level klog.Level
+
+	if conf.LogLevel != "" {
+		if err = level.Set(conf.LogLevel); err != nil {
+			klog.Warningf("failed to set klog log level to %s: %v", conf.LogLevel, err)
+		}
+	}
+	if conf.LogFile != "" {
+		var file *os.File
+
+		if _, err = os.Stat(filepath.Dir(conf.LogFile)); os.IsNotExist(err) {
+			dir := filepath.Dir(conf.LogFile)
+			if err = os.MkdirAll(dir, 0755); err != nil {
+				klog.Warningf("failed to create logfile directory %s (%v).", dir, err)
+				return
+			}
+		}
+		file, err = os.OpenFile(conf.LogFile, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0660)
+		if err != nil {
+			klog.Warningf("failed to open logfile %s (%v).", conf.LogFile, err)
+			return
+		}
+		klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
+		klog.InitFlags(klogFlags)
+		if err := klogFlags.Set("logtostderr", "false"); err != nil {
+			klog.Warningf("Error setting klog logtostderr: %v", err)
+		}
+		if err := klogFlags.Set("alsologtostderr", "true"); err != nil {
+			klog.Warningf("Error setting klog alsologtostderr: %v", err)
+		}
+		klog.SetOutput(file)
```

Comment:
I am unsure of how the shim gets executed again, which might make this comment less relevant or not. 

The only concern I have is that I think `klog` has global pointer for it's logging. This means that I think there might be a conflict `klog.SetOutput` in `config.go`, unless of course the shim is forked into its own process and runs as a standalone server (which was what I don't remember exactly)?

Reply:
@alexanderConstantinescu CNI Shim is a separate process see my diagram